src/codegen/compile.ml

```diff
@@ -10085,6 +10085,10 @@ and fill_pat env ae pat : patternCode =
   PatCode.with_region pat.at @@
   match pat.it with
   | WildP -> CannotFail (G.i Drop)
+  | OptP { it = (WildP | TupP []); _ } ->
```

Comment:
Instead of pattern matching on the inner, can you define a predicate over patterns with a clear semantic in the name (`irrefutable_nonbinding`), and use it here in an `if` clause?

Reply:
Yep, that's a TODO, and it is coming.

Reply:
Done in c06a0c9af

Reply:
Wait the predicate doesn't work?

Reply:
found my oops :-)